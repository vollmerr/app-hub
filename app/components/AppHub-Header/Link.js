import React from 'react';
import styled from 'styled-components';
import { CommandButton, ElementType } from 'office-ui-fabric-react/lib/Button';
import { Link as RouterLink } from 'react-router-dom';

import theme from 'utils/theme';
import PropTypes from 'prop-types';
// import HeaderItem from 'components/Header/HeaderItem';
import Item from './Item';

const LinkStyled = styled(RouterLink) `
  text-decoration: none;
`;

const A = LinkStyled.withComponent('a');

const ButtonStyled = styled(CommandButton) `
  color: ${theme.themePrimary};
  display: block;
  vertical-align: middle;
  border-width: 0;
  margin: 0;
  padding: 0 15px;
  min-width: ${theme.hub.headerHeight};
  height: ${theme.hub.headerHeight};
  box-sizing: border-box;
  text-align: center;
  line-height: normal;
  overflow: visible;
  transition: background-color .467s cubic-bezier(.1,.9,.2,1) 34ms;
  pointer-events: all;
  font-size: ${(props) => props.size || '28px'};
`;

class Link extends React.PureComponent {
  render() {
    const { panel, title, iconProps, onClick, to, href, text, children } = this.props;

    const isLink = onClick || to || href;

    const Button = () => (
      <ButtonStyled
        onClick={onClick ? () => onClick(panel) : null}
        iconProps={iconProps}
        title={title}
        ariaLabel={title}
      >
        {text}
        {children}
      </ButtonStyled>
    );

    // handle internal routing
    if (to) {
      return (
        <Item isLink={isLink}>
          <LinkStyled to={to} >
            <Button />
          </LinkStyled>
        </Item>
      );
    } else if (href) {
      return (
        <Item isLink={isLink}>
          <A href={href} >
            <Button />
          </A>
        </Item>
      );
    }

    // no links
    return (
      <Item isLink={isLink}>
        <Button />
      </Item>
    );
  }
}

Link.propTypes = {
  panel: PropTypes.string,
  title: PropTypes.string,
  iconProps: PropTypes.object,
  onClick: PropTypes.func,
  text: PropTypes.string,
  to: PropTypes.string,
  href: PropTypes.string,
  children: PropTypes.node,
};

export default Link;
