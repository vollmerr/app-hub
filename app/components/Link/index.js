/**
*
* Link
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { Link as RouterLink } from 'react-router-dom';
import styled from 'styled-components';

export const StyledLink = styled(RouterLink) `
  text-decoration: none;
`;

export const A = styled.a`
  text-decoration: none;
`;

class Link extends React.PureComponent {
  render() {
    const { to, href, children, ...props } = this.props;

    if (to) {
      return (
        <StyledLink to={to} {...props}>
          {children}
        </StyledLink>
      );
    }

    return (
      <A href={href} target={'_blank'} rel={'noopener noreferrer'} {...props}>
        {children}
      </A>
    );
  }
}

Link.propTypes = {
  to: PropTypes.string,
  href: PropTypes.string,
  children: PropTypes.node,
};

export default Link;
