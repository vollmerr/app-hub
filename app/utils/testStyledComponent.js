import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

function testStyledComponent(Component, extend = null) {
  describe(`<${Component.displayName} />`, () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Component />);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it(`should render an <${Component.target}> tag`, () => {
      expect(wrapper.type()).toEqual(Component.target);
    });

    it('should have a className attribute', () => {
      expect(wrapper.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      wrapper.setProps({ id });
      expect(wrapper.prop('id')).toEqual(id);
    });

    if (extend) {
      it(`should extend ${extend.name}`, () => {
        expect(wrapper.find(extend).exists()).toEqual(true);
      });
    } else {
      it('should not adopt an invalid attribute', () => {
        const attribute = 'test';
        wrapper.setProps({ attribute });
        expect(wrapper.prop('attribute')).toBeUndefined();
      });
    }
  });
}

export default testStyledComponent;

