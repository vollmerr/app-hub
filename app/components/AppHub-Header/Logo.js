import React from 'react';
import styled from 'styled-components';

import theme from 'utils/theme';

import Link from './Link';
import odiLogo from './odiLogo.png';

const Img = styled.img`
  height: calc(${theme.hub.headerHeight} - 10px);
  margin:  0 0 0 -10px;
`;

const Logo = () => (
  <Link href={'http://www.cdt.ca.gov'}>
    <Img
      src={odiLogo}
      alt={'CDT ODI Logo'}
    />
  </Link>
);

export default Logo;
