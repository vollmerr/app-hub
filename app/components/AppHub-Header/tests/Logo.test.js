import React from 'react';
import { shallow } from 'enzyme';
import testStyledComponent from 'utils/testStyledComponent';

import Logo, { Img } from '../Logo';
import Link from '../Link';

testStyledComponent(Img);


describe('<Logo />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Logo />);
  });

  it('should render the logo as an image', () => {
    expect(wrapper.find(Img).length).toEqual(1);
    expect(wrapper.find(Img).prop('alt')).toMatch(/.+/);
    expect(wrapper.find(Img).prop('src')).toMatch(/.+/);
  });

  it('should be an external link', () => {
    expect(wrapper.find(Link).length).toEqual(1);
    expect(wrapper.find(Link).prop('href')).toMatch(/.+/);
    expect(wrapper.find(Link).prop('title')).toMatch(/.+/);
  });
});
