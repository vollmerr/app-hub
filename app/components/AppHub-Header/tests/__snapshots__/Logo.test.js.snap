// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo__Img /> should render correctly 1`] = `
.c0 {
  height: 50px;
  margin: 0 10px;
}

<img
  className="c0"
/>
`;
